impl11/ddraw/DeviceResources.cpp

```diff
@@ -887,6 +889,8 @@ HRESULT DeviceResources::RenderMain(char* src, DWORD width, DWORD height, DWORD
 	ID3D11Texture2D* tex = nullptr;
 	ID3D11ShaderResourceView* texView = nullptr;
 
+	this->_d3dAnnotation->BeginEvent(L"RenderMain");
```

Comment:
For all calls of `this->_d3dAnnotation` If the `QueryInterface` failed then `_d3dAnnotation` may not be valid.